src/arrayMethodFilter.js

```diff
@@ -5,7 +5,12 @@
  */
 function applyCustomFilter() {
   [].__proto__.filter2 = function(callback) {
-    // write code here
+    let result = [];
+    this.map((el, i) => {
+      return callback(el, i, this)
+        ? result.push(el) : result;
```

Comment:
It is super not obvious why the `result` is returned here. If you don't need `else` block use just `if (condition) {... code if true}`